java/arcs/core/analysis/Deduction.kt

```diff
@@ -0,0 +1,124 @@
+package arcs.core.analysis
+
+import arcs.core.data.expression.Expression
+
+/** Field [Identifier]. Lists of [Identifier]s imply an AccessPath.*/
+typealias Identifier = String
+
+/**
+ * [Deduction]s represent how inputs of a Paxel [Expression] contribute to its outputs.
+ *
+ * [Deduction]s recursively associate identifiers to field paths. These relationships can be
+ * directly translated into [Claim]s and [AccessPath]s.
+ */
+sealed class Deduction {
+
+  /** Unwrap [Deduction] object to get an underlying [Equal] object. */
+  abstract fun unwrap(): Equal
```

Comment:
It seems like unwrapping is no longer necessary -- I'll get rid of it after this meeting.